ethers-signers/src/trezor/mod.rs

```diff
@@ -25,7 +25,12 @@ impl Signer for TrezorEthereum {
 
     /// Signs the transaction
     async fn sign_transaction(&self, message: &TypedTransaction) -> Result<Signature, Self::Error> {
-        self.sign_tx(message).await
+        let mut tx_with_chain = message.clone();
+        if tx_with_chain.chain_id() == None {
```

Comment:
```suggestion
        if tx_with_chain.chain_id().is_none() {
```

Also looks like the clone is only necessary if no chain id is set, so we can move the clone in the if branch instead and execute `sign_tx` in both branches but with different messages.
Same goes for the other calls

Reply:
in the interest of fixing this bug asap, and given people don't really sign multiple txs in a row for this to be a perf issue, let's do the clone stuff as a follow up PR if necessary